cpp/mrc/src/internal/control_plane/client.hpp

```diff
@@ -150,6 +198,8 @@ class Client final : public resources::PartitionResourceBase, public Service
     void request_update();
 
   private:
+    AsyncEventStatus write_event(protos::Event event, bool await_response = false);
```

Comment:
Suggestion: Rename to `write_event_async` if it is an async function as indicated by the return type.